src/Nethermind/Nethermind.Runner/Ethereum/Steps/StartRpc.cs

```diff
@@ -30,8 +32,34 @@ public StartRpc(INethermindApi api)
         public async Task Execute(CancellationToken cancellationToken)
         {
             IJsonRpcConfig jsonRpcConfig = _api.Config<IJsonRpcConfig>();
+            IKeyStoreConfig keyStoreConfig = _api.Config<IKeyStoreConfig>();
             ILogger logger = _api.LogManager.GetClassLogger();
 
+
+            // Update the JWT secret path based on the data directory.
+            try
+            {
+                string defaultPath = "keystore/jwt-secret";
+                string newPath = Path.Combine(keyStoreConfig.KeyStoreDirectory, "jwt-secret");
+                if (string.IsNullOrEmpty(jsonRpcConfig.JwtSecretFile))
+                {
+                    logger.Warn("JsonRpcConfig JwtSecretFile is null or empty because of omission or human error.");
+                    // check if jwt-secret file already exists in previous default directory
+                    if (File.Exists(defaultPath))
+                    {
+                        // move the jwt-secret file
+                        logger.Warn($"jwt-secret already exists at {defaultPath}. Moving it to {newPath} as data directory has been updated");
```

Comment:
a side note: if you intend to take corrective action then they are **info** logs and not **warning** logs.

for now until we decide if we need to move or not based on @MarekM25 response, 
we can just leave it as is.